sql/sql_partition.cc

```diff
@@ -5929,6 +5929,23 @@ the generated partition syntax in a correct manner.
           *partition_changed= true;
         }
       }
+
+      // In case of PARTITION BY KEY(), check if primary key has changed
+      // System versioning also implicitly adds/removes primary key parts
+      if (alter_info->partition_flags == 0 && part_info->list_of_part_fields
+          && part_info->part_field_list.elements == 0)
+      {
+        if (alter_info->flags & (ALTER_DROP_SYSTEM_VERSIONING | ALTER_ADD_SYSTEM_VERSIONING))
+          *partition_changed= true;
+
```

Comment:
add `else if (!*partition_changed)`

Reply:
did it not exactly that way; see the code

Reply:
Is it fixed?